tests/behat/features/bootstrap/LibContext.php

```diff
@@ -0,0 +1,1399 @@
+  }
+
+  /**
+   * Step
+   *
+   * @Then a :relationType entry is not shown
+   */
+  public function findRelationTypeEntryNotShown($relType) {
+    $this->check($this->objectPage->entryIsNotShown($relType));
+  }
+
+  /**
+   * Checks for the reserve-button being shown and visible
+   *
+   * @Then it is possible to click to reserve the material
+   */
+  public function findReserveMaterialButton() {
+    $this->check($this->objectPage->hasReservationButton());
+  }
+
+  /**
+   * Step
+   *
+   * @Then I can see :title in the search results first page
+   */
+  public function findTitleInTheSearchResultsFirstPage($title) {
+    $title = $this->translateArgument($title);
+
+    $this->check($this->searchPage->findTitleOnPage($title));
+  }
+
+  /**
+   * GetPage - quick reference to the getPage element. Makes code more readable.
+   *
+   * @return \Behat\Mink\Element\DocumentElement
+   *   returns a page element
+   */
+  public function getPage() {
+    return $this->minkContext->getSession()->getPage();
+  }
+
+  /**
+   * Step
+   *
+   * @Then I am prompted to login
+   */
+  public function getPromptToLogin() {
+     $this->check($this->objectPage->getPromptToLogin());
+  }
+
+  /**
+   * Navigate to a page.
+   *
+   * @todo should only navigate if the path is different from the current.
+   *
+   * @param string $path
+   *   The path to navigate to.
+   */
+  public function gotoPage($path) {
+    $this->minkContext->visit($path);
+  }
+
+
+  /**
+   * Go to the search page.
+   *
+   * @param string $string
+   *   String to search for.
+   *
+   * @throws Exception
+   *   In case of error.
+   *
+   * @Given I have searched for :string
+   */
+  public function gotoSearchPage($string) {
+    // First we try to translate the argument, to see if there's anything we should pick out first.
+    $searchString = $this->translateArgument($string);
+    $this->logMsg(($this->searchPage->getVerboseSearchResult() == "on"), "Searches for " . urlencode($searchString) . "\n");
+    $this->lastSearchString = $searchString;
+    $this->gotoPage('/search/ting/' . urlencode($searchString));
+  }
+
+  /**
+   * @Given I create files :mfile from opensearch on relation :relation
+   *
+   * @Param string $mfile
+   *    Filename basis for the two files being created, appended with _pos.mat and _neg.mat
+   * @Param string $relation
+   *    The relation we are looking for, in opensearch terminology, f.ex. dbcaddi:hasCreatorDescription
+   */
+  public function ICreateFilesForRelation($mfile, $relation) {
+    for ($i = 1 ; $i < 12000; $i = $i + 50) {
+      $this->ICreateFilesForRelationChunk($mfile, $relation, $i);
+    }
+  }
+
+  /**
+   * Helper function for ICreateFilesForRelation.
+   *
+   * @param string $mfile
+   *    Filename base for the two files being created.
+   * @param string $relation
+   *    The relation we are looking for.
+   * @param integer $start
+   *    The starting point for the service retrieval.
+   */
```

Comment:
@throws ErrorException

Reply:
I added it, although my phpstorm does not think that exception is thrown in it.